pages/index.tsx

```diff
@@ -1,30 +1,116 @@
 import React from 'react'
+import camelCase from 'lodash/camelCase'
+import { documentToReactComponents } from '@contentful/rich-text-react-renderer'
 
-import { fetchEntries } from '../services/contentful'
+import { fetchContent } from '../services/contentful'
 import { LayoutFramework } from '../components/layouts/Framework'
-import { Post } from '../components/Post'
-import { PostType } from '../common'
+import { Hero } from '../components/presenters/Hero'
+import { ContentPanel, LeftCol } from '../components/presenters/ContentPanel'
 
-export default function Home({ posts }) {
-  return (
-    <LayoutFramework>
-      {posts.map(({ title }: PostType) => {
-        return <Post title={title} />
-      })}
-    </LayoutFramework>
-  )
+export type ContentType =
+  | 'Hero'
+  | 'ContentBlock'
+  | 'LiveExample'
+  | 'CodeBlock'
+  | 'APITable'
+
+interface HomeProps {
+  sections: any
 }
 
 export async function getStaticProps() {
-  const res = await fetchEntries()
-
-  const posts = await res.map((post: any) => {
-    return post.fields
-  })
+  const response = await fetchContent(`
+    {
+      sectionCollection(limit: 10) {
+        items {
+          title
+          articleCollection(limit: 10) {
+            items {
+              ... on Article {
+                contentCollection(limit: 10) {
+                  items {
+                    __typename
+                    ... on Hero {
+                      title
+                      heroDescription: description
+                    }
+                    ... on ContentBlock {
+                      title
+                      description {
+                        json
+                      }
+                    }
+                    ... on LiveExample {
+                      title
+                      description {
+                        json
+                      }
+                    }
+                    ... on CodeBlock {
+                      title
+                      description {
+                        json
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `)
 
   return {
     props: {
-      posts,
+      sections: response.sectionCollection.items,
     },
   }
 }
+
+/**
+ * Map content types to components and fields to props
+ *
+ */
+function renderPresenter(type: ContentType, fields: any): React.ReactElement {
+  const id = camelCase(fields.title)
+
+  const componentMap = {
+    Hero: (
```

Comment:
These could be 'adapters' defined alongside the presentational components themselves.

Reply:
Maybe the Adapters should sit in `pages/PageName/adapters`?